src/components/Migration/MigrationModal.tsx

```diff
@@ -240,17 +272,10 @@ function MigrationModal({ open, handleClose }: { open: boolean; handleClose: any
                         <Button
                           variant="outlined"
                           onClick={() => onSeekApproval(row.initialAsset)}
-                          disabled={isPendingTxn(
-                            pendingTransactions,
-                            `approve_migration_${row.initialAsset.toLowerCase()}`,
-                          )}
+                          disabled={isPendingTxn(pendingTransactions, pendingTransactionDispatchType(row))}
                         >
                           <Typography>
-                            {txnButtonText(
-                              pendingTransactions,
-                              `approve_migration_${row.initialAsset.toLowerCase()}`,
-                              t`Approve`,
-                            )}
+                            {txnButtonText(pendingTransactions, pendingTransactionDispatchType(row), t`Approve`)}
```

Comment:
Transaction type for approve is not different between hasDust=true/false

Reply:
Fixed 👍